pkg/apis/crd/v1alpha2/types.go

```diff
@@ -299,13 +299,25 @@ type IPPool struct {
 	Status IPPoolStatus `json:"status"`
 }
 
+// SubnetIPRange is a set of contiguous IP addresses belonging to same subnet, or a whole subnet as
+// specified in Gateway.
+type SubnetIPRange struct {
+	// The CIDR of the Subnet Gateway, e.g. 10.10.20.1/24.
+	Gateway string `json:"gateway"`
```

Comment:
Do you mean you want this single field for both gateway and prefix length? Maybe revise the comments to make it more clearer then.

Originally, I hoped to reuse the IPRange definition which might enable us to share code with other IP allocation cases (when allocating an IP from a pool but without gateway or VLAN). But I got your idea of a separate struct with subnet fields. No strong opinion here.

Reply:
I can split gateway from prefix length if its more intuitive.
I hoped to reuse IPRange as well, but it seems a bit forced with attribute differences..

Reply:
I am fine with either way. Maybe others have an opinion.

Reply:
I feel that having the prefix length being separate from the gateway IP is more intuitive. Having a "/" prefix following the gateway is something pretty unusual.

Reply:
Done

Reply:
I just thought of another problem - we discussed the case that we divide a big subnet to smaller CIDRs, and assign them to Namespaces, in which case all these CIDRs share the subnet, gateway, and VLAN. So, with the current definition, we can use only Start - End to express such a CIDR. It works too, but I like to check your opinion whether you think a separate CIDR field (only one of CIDR or Start+End should be set) works better.

Reply:
This makes sense, and allows to reuse the `IPRange` struct. However I feel that configuration is becoming a bit complex. I think we should require the user to specify both the range (start/end pair or cidr) and gateway+prefix, otherwise deriving defaults automatically may be confusing.

Reply:
Yes, I also feel adding a separate CIDR field sounds better. No preference we should reuse IPRange or not. Originally I thought we can add extra fields (VLAN, gateway, prefixLen) to IPRange, and then we can share more IP allocation code with other cases. If we anyway need a new SubnetIPRange struct, reusing IPRange in the struct does not help code sharing.